monai/transforms/intensity/array.py

```diff
@@ -2301,3 +2302,41 @@ def __call__(self, image: NdarrayOrTensor):
 
         mask = np.stack(foregrounds).all(axis=0)
         return convert_to_dst_type(src=mask, dst=image)[0]
+
+
+class ComputeHoVerMaps(Transform):
+    """Compute horizontal and vertical maps from an instance mask
+    It generates normalized horizontal and vertical distances to the center of mass of each region.
+    Args:
+        dtype: the data type of output Tensor. Defaults to `"float32"`.
+
+    Return:
+        A torch.Tensor with the size of [2xHxW[xD]], which is stack horizontal and vertical maps
+
+    """
+
+    def __init__(self, dtype: DtypeLike = "float32") -> None:
+        super().__init__()
+        self.dtype = dtype
+
+    def __call__(self, mask: NdarrayOrTensor):
+        instance_mask = convert_data_type(mask, np.ndarray)[0]
+
+        h_map = instance_mask.astype(self.dtype)
+        v_map = instance_mask.astype(self.dtype)
```

Comment:
These two maps might share the same memory, the in-place changes in line 2338/2339 might be incorrect

Reply:
Good catch! Right, if the the instance_mask has the same type as output masks, there won't be any copy. will fix it. Thank